src/wasm/wasm-binary.cpp

```diff
@@ -71,8 +71,9 @@ int32_t WasmBinaryWriter::writeU32LEBPlaceholder() {
   return ret;
 }
 
-void WasmBinaryWriter::writeResizableLimits(Address initial, Address maximum, bool hasMaximum) {
-  uint32_t flags = hasMaximum ? 1 : 0;
+void WasmBinaryWriter::writeResizableLimits(Address initial, Address maximum,
+                                            bool hasMaximum, bool shared) {
+  uint32_t flags = (uint32_t) hasMaximum | (uint32_t) shared << 1;
```

Comment:
Instead of two bools in the signature, would it make sense to have a `MemoryFlags` enum?
```c++
enum MemoryFlags {
  memoryHasMaximum = 1 << 0,
  memoryIsShared = 1 << 1
};
```
As it is now we're basically inlining the knowledge of the flag layout, which is probably less obvious in the future.
Failing that a link in a comment to the threads proposal might be good, because otherwise that's implicit knowledge.